operators/src/util/raster_stream_to_geotiff.rs

```diff
@@ -0,0 +1,231 @@
+use core::slice;
+use futures::StreamExt;
+use gdal::{
+    raster::{Buffer, GdalType},
+    Driver,
+};
+use gdal_sys::{VSIFree, VSIGetMemFileBuffer};
+use geoengine_datatypes::{
+    primitives::{SpatialBounded, TimeInterval},
+    raster::{Blit, GeoTransform, Grid2D, GridShape2D, GridSize, Pixel, RasterTile2D},
+    spatial_reference::SpatialReference,
+};
+use std::{
+    convert::TryInto,
+    sync::mpsc::{Receiver, Sender},
+};
+use std::{ffi::CString, sync::mpsc};
+
+use crate::util::Result;
+use crate::{
+    engine::{QueryContext, QueryRectangle, RasterQueryProcessor},
+    error::Error,
+};
+
+pub async fn raster_stream_to_geotiff_bytes<T, C: QueryContext + 'static>(
+    processor: Box<dyn RasterQueryProcessor<RasterType = T>>,
+    query_rect: QueryRectangle,
+    query_ctx: C,
+    width: u32,
+    height: u32,
+    no_data_value: Option<f64>,
+    spatial_reference: SpatialReference,
+) -> Result<Vec<u8>>
+where
+    T: Pixel + GdalType,
+{
```

Comment:
add a guard to check that the tile is <= a (config) constant of tiles? i guess we can use 2x2 tiles for now?

Reply:
I added a parameter c0a6bbafe5162cabd6d41aabc1816a02ddc1dc82 . I will add a config parameter in the WCS PR, because I can only test it there (operators crate doesn't know about config)

Reply:
thats fine for me :)